diff --git a/node_modules/@greenwood/plugin-adapter-aws/src/index.js b/node_modules/@greenwood/plugin-adapter-aws/src/index.js
index 5c6c234..eb7115d 100644
--- a/node_modules/@greenwood/plugin-adapter-aws/src/index.js
+++ b/node_modules/@greenwood/plugin-adapter-aws/src/index.js
@@ -2,6 +2,20 @@ import fs from "node:fs/promises";
 import path from "node:path";
 import { checkResourceExists } from "@greenwood/cli/src/lib/resource-utils.js";
 
+function getSsrRoutes(compilation) {
+  const { config, graph } = compilation;
+
+  return graph.filter((page) => {
+    let is = page.isSSR && !page.staticPaths && page.prerender !== true;
+
+    if (is && config.prerender && page.prerender !== false) {
+      is = false;
+    }
+
+    return is;
+  });
+}
+
 // https://docs.aws.amazon.com/lambda/latest/dg/services-apigateway.html#apigateway-example-event
 // https://docs.aws.amazon.com/lambda/latest/dg/urls-invocation.html
 function generateOutputFormat(id, type, segmentKey) {
@@ -77,7 +91,7 @@ async function awsAdapter(compilation) {
   const { outputDir, projectDirectory } = compilation.context;
   const { basePath } = compilation.config;
   const adapterOutputUrl = new URL("./.aws-output/", projectDirectory);
-  const ssrPages = compilation.graph.filter((page) => page.isSSR);
+  const ssrPages = getSsrRoutes(compilation); // compilation.graph.filter((page) => page.isSSR);
   const apiRoutes = compilation.manifest.apis;
 
   if (await checkResourceExists(adapterOutputUrl)) {
