diff --git a/node_modules/@greenwood/plugin-css-modules/src/index.js b/node_modules/@greenwood/plugin-css-modules/src/index.js
index 2bb251f..499c400 100644
--- a/node_modules/@greenwood/plugin-css-modules/src/index.js
+++ b/node_modules/@greenwood/plugin-css-modules/src/index.js
@@ -184,11 +184,13 @@ class ScanForCssModulesResource {
     }
   }
 
-  async shouldIntercept(url) {
+  async shouldIntercept(url, request, response) {
     const { pathname, protocol } = url;
+    const isHtmlResponse = response.headers.get("Content-Type")?.includes("text/html");
 
     return (
-      url.pathname.endsWith("/") || (protocol === "file:" && pathname.endsWith(this.extensions[0]))
+      (pathname.endsWith("/") && isHtmlResponse) ||
+      (protocol === "file:" && pathname.endsWith(this.extensions[0]))
     );
   }
 
@@ -236,7 +238,11 @@ class ScanForCssModulesResource {
       `,
       );
 
-      return new Response(newBody);
+      return new Response(newBody, {
+        headers: response.headers,
+        status: response.status,
+        statusText: response.statusText,
+      });
     } else if (protocol === "file:" && pathname.endsWith(this.extensions[0])) {
       // handle this primarily for SSR / prerendering use case
       const cssModulesMap = fs.readFileSync(
