diff --git a/node_modules/wc-compiler/src/jsx-loader.js b/node_modules/wc-compiler/src/jsx-loader.js
index 2e24c74..96545c3 100644
--- a/node_modules/wc-compiler/src/jsx-loader.js
+++ b/node_modules/wc-compiler/src/jsx-loader.js
@@ -166,6 +166,7 @@ function parseJsxElement(
             }
 
             // <button onclick={(e: Event) => this.count.set(this.count.get() * 2)}>Double (++)</button>
+            // TODO: handle private, e.g. this.#someFunction?
             if (expression.type === 'ArrowFunctionExpression' && expression.body) {
               // quick hack to get expression contents until we can properly build this all up from an AST
               const contents = generate(expression.body);
diff --git a/node_modules/wc-compiler/src/jsx.d.ts b/node_modules/wc-compiler/src/jsx.d.ts
index ca6e96a..349c2ed 100644
--- a/node_modules/wc-compiler/src/jsx.d.ts
+++ b/node_modules/wc-compiler/src/jsx.d.ts
@@ -7,6 +7,21 @@ type ElementAttributes<E extends HTMLElement> = {
   [A in keyof E]?: E[A] extends (...args: any) => any ? any : IsCSSStyleDeclaration<E[A]>;
 } & {
   class?: string;
+  children?: any;
+};
+
+// creates a utility type for SVG attributes, allowing string values for SVG-specific attributes
+type SVGElementAttributes<E extends SVGElement> = {
+  [A in keyof E]?: string | undefined;
+} & {
+  class?: string;
+  fill?: string;
+  stroke?: string;
+  d?: string;
+  viewBox?: string;
+  xmlns?: string;
+  preserveAspectRatio?: string;
+  [key: string]: any;
 };
 
 type PopoverState = 'auto' | 'manual' | 'hint';
@@ -25,9 +40,15 @@ type PopoverAttributes = {
 // map each HTML tag to its attributes
 type IntrinsicElementsFromDom = {
   [E in keyof HTMLElementTagNameMap]: ElementAttributes<HTMLElementTagNameMap[E]> &
-    (E extends 'button' | 'input' ? PopoverAttributes : {});
+    (E extends 'button' | 'input' ? PopoverAttributes : {}) &
+    (E extends 'button' ? { tabindex?: number | string } : {});
+};
+
+// map each SVG tag to its attributes
+type IntrinsicElementsFromSVG = {
+  [E in keyof SVGElementTagNameMap]: SVGElementAttributes<SVGElementTagNameMap[E]>;
 };
 
 declare namespace JSX {
-  interface IntrinsicElements extends IntrinsicElementsFromDom {}
+  interface IntrinsicElements extends IntrinsicElementsFromDom, IntrinsicElementsFromSVG {}
 }
\ No newline at end of file
